### PR TITLE
Add annotations from expect entity to actual entity. Required after KT-58551

### PR DESCRIPTION
**Pull request by woainikk**
[KT-58551](https://youtrack.jetbrains.com/issue/KT-58551)

### CODE REVIEW

**Review by shanshin — APPROVED** [review]

File: formats/json/nativeMain/src/kotlinx/serialization/json/internal/FormatLanguage.kt [hunk 1]
```diff
@@ -7,6 +7,16 @@ package kotlinx.serialization.json.internal;
...
+@Target(
+    AnnotationTarget.FUNCTION,
+    AnnotationTarget.PROPERTY_GETTER,
+    AnnotationTarget.PROPERTY_SETTER,
+    AnnotationTarget.FIELD,
+    AnnotationTarget.VALUE_PARAMETER,
+    AnnotationTarget.LOCAL_VARIABLE,
+    AnnotationTarget.ANNOTATION_CLASS
+)
 public actual annotation class FormatLanguage(
```
Comment (sandwwraith) [inline formats/json/nativeMain/src/kotlinx/serialization/json/internal/FormatLanguage.kt:20]: There's an `actual annotation class FormatLanguage` in JS source set, too

Also, what does the new rule says about `actual typealias`? We have this in JVM

**Review by sandwwraith — CHANGES_REQUESTED** [review]
.

File: formats/json/nativeMain/src/kotlinx/serialization/json/internal/FormatLanguage.kt (hunk 1 above)
Comment (woainikk) [reply formats/json/nativeMain/src/kotlinx/serialization/json/internal/FormatLanguage.kt:20]: Fixed for Js source set.
"Annotations with retention SOURCE on expect are not checked if actual is typealias to compiled code." - and there is no compiler errors except error with FormatLanguage

Comment (sandwwraith) [reply formats/json/nativeMain/src/kotlinx/serialization/json/internal/FormatLanguage.kt:20]: But `kotlin.Target` has default retention (runtime)

**Review by sandwwraith — APPROVED** [review]

**Pull request merged**